pkg/apis/configuration/validation/validation_test.go

```diff
@@ -1437,3 +1437,73 @@ func TestValidateUpstreamLBMethodFails(t *testing.T) {
 		}
 	}
 }
+
+func createPointerFromInt(n int) *int {
+	return &n
+}
+
+func TestValidatePositiveInt(t *testing.T) {
+	tests := []struct {
+		number *int
+		msg    string
+	}{
+		{
+			number: nil,
+			msg:    "valid (nil)",
+		},
+		{
+			number: createPointerFromInt(0),
+			msg:    "valid (0)",
+		},
+		{
+			number: createPointerFromInt(1),
+			msg:    "valid (1)",
+		},
+	}
+
+	for _, test := range tests {
+		allErrs := validatePositiveInt(test.number, field.NewPath("int-field"))
+
+		if len(allErrs) != 0 {
+			t.Errorf("validatePositiveInt returned errors for case: %v", test.msg)
+		}
+	}
+}
+
+func TestValidatePositiveIntFails(t *testing.T) {
+	tests := []struct {
```

Comment:
since there is only one set of input arguments here, having an array of tests as well as the loop `for _, test := range tests` seems redundant.